pkg/query/storeset.go

```diff
@@ -128,12 +129,21 @@ func NewStoreSet(
 		Name: "thanos_store_nodes_grpc_connections",
 		Help: "Number indicating current number of gRPC connection to store nodes. This indicates also to how many stores query node have access to.",
 	})
+	storeStatus := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "thanos_store_status",
+		Help: "The status of stores as indicated on the /stores page. 1 is UP 0 is DOWN",
+	},
+		[]string{
+			"address",
```

Comment:
That's the main problem we want to avoid. Putting potentially dynamic addresses to the metrics which increases cardinality.

What's wrong again with:
```
ar nodeInfoDesc = prometheus.NewDesc(
	"thanos_store_node_info",
	"Number of nodes with the same external labels identified by their hash. If any time-series is larger than 1, external label uniqueness is not true",
	[]string{"external_labels"}, nil,
)
```
It has exactly the same logic you have given here just encoded bit vaguely in collector logic - it might be not clear. Anyway in my opinion the problem must be somewhere else.. Maybe the cluster was down for a short time?

However, I am happy with you metric that will **replace** `thanos_store_node_info` as we can improve name and help. It's not only meta info - it also gives you healthiness info. However I would stick to `external_labels` - I think even for your use case it's just enough to have external labels, right?

Reply:
The problem with `thanos_store_node_info ` is:

1. its not a bool value, which means it becomes a little more complex to eval simply "can this query talk to a store with labels set X"
2. it records how many stores are available with the same label set
3. If a store becomes unhealthy, its removed from the list of healthy stores which means in the re-construction of the list, it won't be exposed in the metric as down which means it will never expose a metric with 0. 
```
externalLabelOccurrencesInStores := map[string]int{}
	for _, st := range healthyStores {
		externalLabelOccurrencesInStores[externalLabelsFromStore(st)]++
	}
```
4. In our setup we have 1 entry point per cluster e.g. `cluster-1.mb.io` and I just want to know if "Global Query" considers this particular store healthy or not.

Reply:
> That's the main problem we want to avoid. Putting potentially dynamic addresses to the metrics which increases cardinality.

I don't see any other solution as the external label set is not always available in case if the first request results in an error. The address is the only constant we have.

Reply:
@bwplotka I've pushed using label set and its hash as labels.